src/main/java/com/couchbase/intellij/tree/PillowFightDialog.java

```diff
@@ -0,0 +1,1507 @@
+                validatePathcountTextField(originalColor);
+            }
+
+            @Override
+            public void removeUpdate(DocumentEvent documentEvent) {
+                validatePathcountTextField(originalColor);
+            }
+
+            @Override
+            public void changedUpdate(DocumentEvent documentEvent) {
+                validatePathcountTextField(originalColor);
+            }
+        });
+
+        init();
+
+            /*
+        } catch (Exception e) {
+            Messages.showErrorDialog("Could not connect to the cluster. Please check your network connectivity, " +
+                    " if the cluster is active or if the credentials are still valid.", "Couchbase Plugin Error");
+            System.err.println(e);
+        }
+             */
+
+        /*
+        try {
+            Cluster cluster = ActiveCluster.getInstance().get();
+            if (cluster==null) {
+                System.err.println("Active cluster is established. Connecting now...");
+                // TODO: Handle multiple saved clusters
+                try {
+                    Map<String, SavedCluster> savedClusters = DataLoader.getSavedClusters();
+                    String savedKey = (String) savedClusters.keySet().toArray()[0];
+                    ActiveCluster.getInstance().connect((SavedCluster) savedClusters.get(savedKey));
+                    cluster = ActiveCluster.getInstance().get();
+                    Messages.showErrorDialog("Could not connect to the cluster. Please check your network connectivity, " +
+                            " if the cluster is active or if the credentials are still valid.", "Couchbase Plugin Error");
+                } catch (Exception e) {
+                    System.err.println("Error" + e);
+                }
+            }
+            BucketManager manager = cluster.buckets();
+            Set<String> bucketNamesSet = manager.getAllBuckets().keySet();
+            String[] bucketNamesArray = bucketNamesSet.toArray(new String[bucketNamesSet.size()]);
+            for (int i = 0; i < bucketNamesArray.length; i++) {
+                bucketComboBox.addItem(bucketNamesArray[i]);
+            }
+        } catch (Exception e) {
+            System.out.println("Error: " + e);
+        }
+         */
+    }
+
+    private void validatePersistToTextField(Color originalColor) {
+        String input = persistToTextField.getText();
+        try {
+            if (input == null || input.trim().isEmpty()) {
+                persistToTextField.setForeground(originalColor);
+                setOKActionEnabled(true);
+                return;
+            }
+            if (Integer.parseInt(input) < -1) {
+                persistToTextField.setForeground(JBColor.RED);
+                setOKActionEnabled(false);
+            } else {
+                persistToTextField.setForeground(originalColor);
+                setOKActionEnabled(true);
+            }
+        } catch (NumberFormatException e) {
```

Comment:
All the fields that should only contain numbers should use "JSpinner" instead. This refactor will be a bit big but it will massively simplify your code. For instance:
JSpinner threadsField = new JSpinner(new SpinnerNumberModel(4, -1, 100, 1));